commit c0a68528675e4350a206099753e03189bba73c21
Author: Jamie Coldhill <wrxtasy@amnet.net.au>
Date:   Sun May 1 02:26:29 2016 +0800

    Reduce rendering fps of the GUI layer when playing videos in fullscreen mode

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index cb10ffa..c87904a 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2831,10 +2831,10 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
     m_skipGuiRender = false;
     int fps = 0;
 
-#if defined(TARGET_RASPBERRY_PI) || defined(HAS_IMXVPU)
+#if defined(TARGET_RASPBERRY_PI) || defined(HAS_IMXVPU)  || defined(HAS_LIBAMCODEC)
     // This code reduces rendering fps of the GUI layer when playing videos in fullscreen mode
     // it makes only sense on architectures with multiple layers
-    if (g_graphicsContext.IsFullScreenVideo() && !m_pPlayer->IsPausedPlayback() && g_renderManager.IsVideoLayer())
+    if (m_pPlayer->IsPlayingVideo() && !m_pPlayer->IsPausedPlayback() && g_renderManager.IsVideoLayer())
       fps = CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_LIMITGUIUPDATE);
 #endif
 
@@ -2847,6 +2847,8 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
     {
       if (!m_skipGuiRender)
         g_windowManager.Process(CTimeUtils::GetFrameTime());
+      else if (!g_graphicsContext.IsFullScreenVideo())
+        g_renderManager.FrameMove();
     }
     g_windowManager.FrameMove();
   }
